### PR TITLE
* Avoid loose Transformable Resources not being updated on database.

### PR DESCRIPTION
**Pull request by ngs-mtech**
We have found that sequences of SingleSessionCommandService executions and the clearing of transactionResouces on commit not owning the transaction looses some Transformable Resouces not being updated on database.

Proposed solution solves the issue.

Here are some logs performed before clearing transactionResources that shows that there are TransformableResources not being merged but cleared:
osed solution solves the issue.

Here are some logs performed before clearing transactionResources that shows that there are TransformableResources not being merged but cleared:

```
2017-11-02 17:40:00,095 WARN  [org.drools.persistence.jta.JtaTransactionManager] (jBPMClusteredScheduler_Worker-5) Missing transactional resouces
2017-11-02 17:40:00,095 WARN  [org.drools.persistence.jta.JtaTransactionManager] (jBPMClusteredScheduler_Worker-5) org.drools.persistence.txsync.container -- org.drools.persistence.TransactionSynchronizationContainer@675d92f2
2017-11-02 17:40:00,095 WARN  [org.drools.persistence.jta.JtaTransactionManager] (jBPMClusteredScheduler_Worker-5) TriggerUpdateTransactionSynchronization-org.drools.persistence.SingleSessionCommandService@2b872177 -- true
2017-11-02 17:40:00,095 WARN  [org.drools.persistence.jta.JtaTransactionManager] (jBPMClusteredScheduler_Worker-5) app-updateable-resource -- [org.jbpm.persistence.processinstance.ProcessInstanceInfo@f070cfd3, org.drools.persistence.info.SessionInfo@34a621f4]
```

### CODE REVIEW

File: drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java [hunk 1]
```diff
@@ -234,9 +235,20 @@ public boolean begin() {
                     logger.warn( "Unable to begin transaction", e);
                     throw new RuntimeException( "Unable to begin transaction", e );
                 }
+            }finally{
```
Comment (baldimir) [inline drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java:235]: Minor formatting problem.

File: drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java [hunk 2]
```diff
@@ -247,9 +259,10 @@ public void commit(boolean transactionOwner) {
                 logger.warn( "Unable to commit transaction", e);
                 throw new RuntimeException( "Unable to commit transaction",
                                             e );
+            }finally{
```
Comment (baldimir) [inline drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java:259]: Minor formatting problem. Please add spaces.

**Review by baldimir — CHANGES_REQUESTED** [review]
Could you please add a reproducer JUnit test modelling the problem? That will help us with regression testing in the future. As of merge or if the solution is correct, @mswiderski could you please take a look?

**Comment by baldimir** [comment]
jenkins execute full downstream build

**Comment by baldimir** [comment]
jenkins execute full downstream build

File: drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java (hunk 2 above)
Comment (ngs-mtech) [reply drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java:259]: fixed

File: drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java (hunk 1 above)
Comment (ngs-mtech) [reply drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/jta/JtaTransactionManager.java:235]: fixed

**Comment by mariofusco** [comment]
jenkins execute full downstream build

**Comment by mariofusco** [comment]
outdated

**Pull request closed without merging**